src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/VsSolutionEventListener.cs

```diff
@@ -117,6 +120,8 @@ public int OnBeforeUnloadProject(IVsHierarchy pRealHierarchy, IVsHierarchy pStub
 
         public int OnAfterOpenSolution(object pUnkReserved, int fNewSolution)
         {
+            IsSolutionClosing = false;
+
             return HResult.NotImplemented;
```

Comment:
Is NotImplemented still the correct return value for these methods? Wouldn't HResult.OK be a more accurate reflection of what the code does?

Reply:
Perhaps. It might make sense to return `HResult.OK` from all of these as `NotImplemented` doesn't appear to receive any special treatment. There seems to be an outage for symbols right now, so I'm unable to look into `msenv.dll` to see the exact behaviour. I've tested extensively with `NotImplemented` so am reluctant to change it now as I'd like to move on to other things.

Reply:
There is always //index: http://index/?query=%60OnAfterOpenSolution&rightProject=VS(Repo)&file=src%5Cenv%5Cmsenv%5Ccore%5Cvsslnpst.cpp&line=21425

Looks like OK might be better, as NotImplemented could cause it to not be fired again, though it actually looks like the handling of the Prioritized handlers makes the normal ones redundant as the return value is ignored from them.

Reply:
I think you'll need to change the design of this slightly; this currently only ever listens to solution events if the project has ".NET" capability, dependencies node can be present without that.

Reply:
@davkean where is this tied to .NET capability?

Reply:
Index shows that the return value only matters for `OnBeforeCloseSolution`. If `NotImplemented` is returned, it won't call again. I've changed them all to `Ok` and tested that things still work correctly.

Reply:
@drewnoakes Look at the caller of StartListeningAsync.

Reply:
@davkean I've updated the PR, moving the service to global scope and initialising it in the package. Please re-review.